dev/io.openliberty.microprofile.telemetry.2.0.logging.internal/src/io/openliberty/microprofile/telemetry20/logging/internal/OpenTelemetryLogHandler.java

```diff
@@ -242,19 +241,14 @@ public Object formatEvent(String source, String location, Object event, String[]
                         } else {
                             attributes.put(extKey, k.getStringValue());
                         }
-
-                        if (extKey.equals("ext_appName")) {
-                            String appName = k.getStringValue();
-                            if (!appName.contains("io.openliberty") && !appName.contains("com.ibm.ws")) {
-                                otelInstance = OpenTelemetryAccessor.getOpenTelemetryInfo(appName).getOpenTelemetry();
-                            }
-                        }
                     }
                 }
             }
 
-            if (otelInstance == null) {
-                otelInstance = this.openTelemetry;
+            if (OpenTelemetryAccessor.isRuntimeEnabled()) {
+                otelInstance = openTelemetry;
+            } else {
+                otelInstance = OpenTelemetryAccessor.getOpenTelemetryInfo().getOpenTelemetry();
```

Comment:
I think the `otelInstance != null` below should have an `&&  otelInstance.isEnabled.` 

Otherwise you can get the no-op from an ErrorOpenTelemetry or similar and waste time using it. 

I would also suggest moving LogRecordBuilder builder = null; and OpenTelemetry otelInstance = null; from further up to around here so you can quickly see they have no prior state before they're set  here.

Reply:
Started using getEnabled. I moved OpenTelemetry otelInstance down but builder needs to be initialized outside of the loop, might be able to clean this up after we refactor a few things soon. Thanks!